### ISSUE TITLE
Elastic agents finishing before all logs are sent to GoCD after upgrade to 2.0.0-113

### ISSUE THREAD

**Issue by cfontes** [gocd/kubernetes-elastic-agents#66]
Hello,

So I've been using this plugin for a while and it generally works great.

After I've upgraded to the latest GoCD and the latest version of this plugin (kubernetes-elastic-agent-2.0.0-113.jar), my EAs have been dying without giving me all the logs from the container

As an example, in GoCD this is the last line of log I get:

`[INFO] [INFO] Compiling 7 source files to /godata/pipelines/EventDocumenter/event-documenter/target/classes****`

And on the gcloud console of the container I have a lot more stuff

```
2018-11-15 15:32:57,277 [stdout] - 2018-11-15 15:32:57,272 INFO  [Thread-24] CommandLine:51 - [INFO] [INFO] Compiling 7 source files to /godata/pipelines/EventDocumenter/event-documenter/target/classes     **<=== LAST LINE I SAW ON GoCD Web Console **
2018-11-15 15:33:01,116 [stdout] - 2018-11-15 15:33:01,099 INFO  [pool-2-thread-1] HttpService:138 - Got back 200 from server
.
.
.
Lot's of other stuff, close to 100 lines of output
.
.
.
2018-11-15 15:33:16,895 [stdout] - 2018-11-15 15:33:16,895 INFO  [Thread-24] CommandLine:51 - [INFO] BUILD FAILURE
2018-11-15 15:33:17,706 [stdout] - 2018-11-15 15:33:17,691 INFO  [scheduler-3] DefaultGoPublisher:99 - Agent [gocd-jdk11-dind-991552d2-b280-4ff3-b533-3314cd061a6a, 10.48.2.117, 816045bb-7ead-44c8-8cd6-2e3fca968cae] is reporting build result [Failed] to Go Server for Build [EventDocumenter/39/release/3/release/616
2018-11-15 15:33:17,844 [stdout] - 2018-11-15 15:33:17,834 INFO  [scheduler-3] DefaultGoPublisher:94 - Agent [gocd-jdk11-dind-991552d2-b280-4ff3-b533-3314cd061a6a, 10.48.2.117, 816045bb-7ead-44c8-8cd6-2e3fca968cae] is reporting status [Completing] to Go Server for Build [EventDocumenter/39/release/3/release/616
2018-11-15 15:33:17,934 [stdout] - 2018-11-15 15:33:17,927 INFO  [scheduler-3] ArtifactsPublisher:69 - Pluggable metadata folder is empty.
2018-11-15 15:33:17,936 [stdout] - 2018-11-15 15:33:17,935 INFO  [scheduler-3] DefaultGoPublisher:104 - Agent [gocd-jdk11-dind-991552d2-b280-4ff3-b533-3314cd061a6a, 10.48.2.117, 816045bb-7ead-44c8-8cd6-2e3fca968cae] is reporting build result [Failed] to Go Server for Build [EventDocumenter/39/release/3/release/616
2018-11-15 15:33:19,533 INFO  [ShutdownThread-5] AgentProcessParentImpl$Shutdown:185 - Shutdown hook invoked. Shutting down Process[pid=69, exitValue="not exited"]
```

Do you guys know what could be the issue here?

I am using a GoCD agent derived from the Debian 9 one.

**Comment by GaneshSPatil** [comment]
Hi @cfontes -- A similar issue has been reported [here](https://github.com/gocd/kubernetes-elastic-agents/issues/64).
We've identified the issue and added a fix for it [here](https://github.com/gocd/gocd/pull/5335). The fix will be available as part of the upcoming GoCD release `v18.11.0`, which is out sometimes next week.

Duplicate of #64

**Comment by cfontes** [comment]
Great, thank you guys!
